site/zenodo_rdm/legacy/deserializers/metadata.py

```diff
@@ -257,7 +257,8 @@ def load_rights(self, obj):
         else:
             # If license does not exist in RDM, it is added as custom
             legacy_license = LEGACY_LICENSES.get(obj)
-            ret = {"title": {"en": legacy_license["title"]}}
+            if legacy_license:
+                ret = {"title": {"en": legacy_license["title"]}}
```

Comment:
It is not so bad that it blows up here, since this is the last chance to set the license, that is required to publish the record in RDM. Could we raise a Marshmallow exception? Looks better than python blowing up. e.g:

```python
if not legacy_license:
    raise ValidationError() 
```

For context, this is how we try to figure out the license:

1) Check if the license is an [alias](https://github.com/zenodo/zenodo-rdm/blob/master/legacy/zenodo_legacy/licenses.py#L19)
2) Try to [map the license](https://github.com/zenodo/zenodo-rdm/blob/35616a18b059c9196d8fe3966d0930e9f5ea242d/legacy/zenodo_legacy/data/legacy_to_rdm_map.json#L4) to one in RDM
3) If not defined yet, fallback to a [legacy license](https://github.com/zenodo/zenodo-rdm/blob/35616a18b059c9196d8fe3966d0930e9f5ea242d/legacy/zenodo_legacy/data/legacy_licenses.json#L4) but added it as a custom one in RDM



This issue appears on Github because users declare their licenses wrongly. E.g. `GPLv3+` is not a valid identifier (check [spdx](https://spdx.org/licenses/)) nor an alias (check legacy Zenodo codebase)